.npm/postinstall.js

```diff
@@ -1,23 +1,55 @@
 require("process");
 const { spawn } = require("child_process");
+const { createWriteStream, unlink } = require("fs");
+const { get } = require("https");
 const { join } = require("path");
 
+const downloadPaths = {
+  'darwin': 'https://raw.githubusercontent.com/Arkweid/lefthook/master/.npm/bin/lefthook-mac',
```

Comment:
This script will always install the latest release from the 'master' branch. 
This will cause problems if we want to do something like `npm i @arkweid/lefhook@1.0.0`

Reply:
Yes. That's bad.

Would you suggest to resolve the packages from the releases/tags URL? `https://github.com/Arkweid/lefthook/releases/download/${version}/lefthook_${version}_${os}_${arch}.gz`?

That raises three more questions:
- How to get the installed version from the postinstall script,
- how to determine the `arch` value, and
- whether to unpack the binary from the package via the postinstall script?

Reply:
OOooohh, and trying out some stuff: How to deflate the windows' zip-files with node? I don't really want to include a `zip`-related node-package just for that specific reason...

Reply:
And another thing I just noticed: GitHub doesn't serve the files directly, but need to follow the redirect that the releases-binaries responses contain.

I'm not saying it's impossible, I'll just need more time to implement a proper solution.

Sorry for spamming, I'm just writing as I go. 😛

Reply:
I'm having troubles finding a good solution for windows: I could rely on just downloading the file to a temporary location, call the (hopefully) installed `zip` binary (I would have to check how to do that on a windows machine) and then remove the zip-package.

This of course is a bit more complicated than the MacOS/Linux version, which only needs to deflate the downloaded `.gz` via the node-internal `zlib` module (and which I have a working prototype for).

Reply:
@Arkweid what do you think? :)
I think we should put [zlib-compatible](https://nodejs.org/api/zlib.html) binaries to the releases folder. 
As for redirects, you can do something like
```
if (response.statusCode === 302) {
       //get a new url from response.headers.location
   } else {
      // load file
};
```

Reply:
@HellSquirrel I updated the PR, it should now fetch the binaries from the releases paths (with redirect-resolve). I was only able to test on linux-x64, but it seems to work alright. (I also adapted the bin-path creation from @levenkov, which makes sense if we not deliver all the binaries in the npm package.)

I will click the "Resolve conversation" button now, because I feel this particular discussion is resolved, feel free to open a new one on the updated `postinstall.js`.